### DISCUSSION TITLE
Could you please add `src` folder inside `@keystone-6/core` package

### DISCUSSION THREAD

**Discussion by pahaz** [keystonejs/keystone#8487] [Feature requests]
So from time to time, you need to check internal keystone details.
But the `@keystone-6/core` does not include the `src` folder.

While the `@keystone-6/auth` have it! And it's really helpful.

<img width="445" alt="image" src="https://user-images.githubusercontent.com/1394025/232251440-e25be79d-d205-4b70-8dde-a6601bd7a424.png">

**Comment by dcousens** [comment]
@pahaz if the other packages have `src`, please open a pull request to add it for `@keystone-6/core`.

**Reply by pahaz** [reply]
It's related to `preconstruct` package. Probably, the main differences between packages related to [files section](https://github.com/keystonejs/keystone/blob/main/packages/core/package.json#L191-L207). 

Can I just drop it? or add `src` inside

**Reply by pahaz** [reply]
https://github.com/keystonejs/keystone/pull/8488 -- similar to others packages

**Reply by dcousens** [reply]
Thanks @pahaz!

**Comment by mmachatschek** [comment]
It should be noted that the package sizes increases if additional source code is added.

Actually I would prefer that the auth package wouldn't include the src as well

**Reply by dcousens** [reply]
@mmachatschek if compressed, this should be minimal. Do we have any numbers?

**Reply by pahaz** [reply]
Why? You can't learn fast without it. From time to time you need to look at some source code to understand what's going on.
You need to have very well documentation in order not to add the source code.

I often develop without the Internet, the presence of the source code allows me to do this.

In my case, it improves Developer Experience and improves the learning curve.

**Reply by mmachatschek** [reply]
I'm ok with adding the src files from a function size point of view on vercel. they calculate a max function size of 50mb compressed and max 250mb uncompressed https://vercel.com/docs/concepts/limits/overview#:~:text=The%20maximum%20size%20for%20a%20Serverless%20Function%20is%2050%20MB%20and%20the%20maximum%20unzipped%20size%20is%20250%20MB

on our end, our rather big codebase has a 20mb compressed size on vercel an 50mb uncompressed (node_modules size)

**Reply by dcousens** [reply]
@mmachatschek are you saying your Keystone deployment bundle is under 50 megabytes uncompressed? :open_mouth:

**Reply by mmachatschek** [reply]
@dcousens seems like 😅 we are currently experimenting with graphql yoga and the custom nextjs backend. this is inspired by the work on this [example](https://github.com/keystonejs/keystone/tree/main/examples/nextjs-keystone) and @borisno2's https://github.com/borisno2/on-the-hill-drama-club

our prisma file has 4k lines and the graphql schema has 16k

this is the current status quo (without prisma JSON protocol and without prisma data proxy)
cold start + initial request takes 15-20sec. We hope this is reduced as soon as we use prisma 4.12 and we're finally able to test this new preview feature (I hope 4.13 has more improvements to serverless envs)

but thats a bit off topic to this discussion 😅 

![image](https://user-images.githubusercontent.com/10237069/232689227-0b0968b8-0b48-4b02-a7ee-68ccfc3a384d.png)
